egs/wsj/s5/steps/nnet3/convert_nnet2_to_nnet3.py

```diff
@@ -0,0 +1,441 @@
+    parser.add_argument("nnet3_dir", metavar="src-nnet3-dir", type=str,
+                        help="")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+
+    if not os.path.exists(args.nnet3_dir):
+        os.makedirs(args.nnet3_dir)
+    if args.tmpdir and not os.path.exists(args.tmpdir):
+        os.makedirs(args.tmpdir)
+
+    return args
+
+class Nnet3Model(object):
+    '''
+    Holds configuration for an Nnet3 model.
+    '''
+    
+    def __init__(self):
+        self.input_dim = -1
+        self.output_dim = -1
+        self.counts = defaultdict(int)
+        self.num_components = 0
+        self.components_read = 0
+        self.config = ""
+        self.transition_model = ""
+        self.components = []
+
+    def AddComponent(self, component, pairs):
+        '''
+        Adds components to the model given a 
+        dictionary of key-value pairs.
+        '''
+        self.components_read += 1
+
+        Component = namedtuple('Component', 'ident component pairs')
+
+        if "<InputDim>" in pairs and self.input_dim == -1:
+            self.input_dim = pairs["<InputDim>"]
+
+        # remove nnet2 specific tokens and catch descriptors
+        if component == '<PnormComponent>' and '<P>' in pairs:
+            pairs.pop('<P>')
+        elif component in SPLICE_COMPONENTS:
+            self.components.append(Component("splice", component, pairs))
+            return
+
+        # format pairs: {'<InputDim>':43} -> {'input-dim':43}
+        pairs = ['{0}={1}'.format(TokenToString(key),pairs[key]) for key in pairs]
+        
+        # keep track of layer type number (e.g. affine3)
+        node_name = NODE_NAMES[component]
+        self.counts[node_name] += 1
+
+        # e.g. affine3
+        ident = node_name + str(self.counts[node_name])
+
+        # <PnormComponent> -> PnormComponent
+        component = component[1:-1]
+
+        self.components.append(Component(ident, component, pairs))
+
+    def WriteConfig(self, filename):
+        '''
+        Write config to filename.
+        '''
+        logger.info('Writing config to {0}'.format(filename))
+
+        self.config = filename
+        with open(filename, 'w') as f:
+            for component in self.components:
+                if component.ident == "splice":
+                    continue
+                config_string = ' '.join(component.pairs)
+
+                f.write('component name={name} type={comp_type} {config_string}\n'.format(name=component.ident, comp_type=component.component, config_string=config_string))
+
+            f.write('\n# Component nodes\n')
+            f.write('input-node name=input dim={0}\n'.format(self.input_dim))
+            previous_component='input'
+            for component in self.components:
+                if component.ident == "splice":
+                    # Create splice string for the next node
+                    previous_component=MakeSpliceString(previous_component, component.pairs['<Context>'])
```

Comment:
Sorry, I had forgot about SpliceMaxComponent - it's not explicitly handled at the minute. What is the equivalent in nnet3? I couldn't find a max-descriptor (e.g. 'Max(Offset, ...)')